demos/signature/EVP_Signature_demo.c

```diff
@@ -0,0 +1,261 @@
+     */
+    sign_context = EVP_MD_CTX_new();
+    if (sign_context == NULL) {
+        fprintf(stderr, "EVP_MD_CTX_new failed.\n");
+        goto cleanup;
+    }
+    /*
+     * Initialize the sign context to use the fetched
+     * sign provider.
+     */
+    if (EVP_DigestSignInit_ex(sign_context, NULL, sig_name,
+                              libctx, NULL, priv_key, NULL) != 1) {
+        fprintf(stderr, "EVP_DigestSignInit_ex failed.\n");
+        goto cleanup;
+    }
+    /*
+     * EVP_MD_CTX_get1_md returns the EVP_MD structure corresponding
+     * to the passed EVP_MD_CTX.
+     */
+    sig_alg = EVP_MD_CTX_get0_md(sign_context);
+    if (sig_alg == NULL) {
+        fprintf(stderr, "EVP_MD_CTX_get1_md returned NULL.");
+        goto cleanup;
+    }
+    /*
+     * EVP_DigestSignUpdate() can be called several times on the same context
+     * to include additional data.
+     */
+    if (EVP_DigestSignUpdate(sign_context, hamlet_1, strlen(hamlet_1)) != 1) {
+        fprintf(stderr, "EVP_DigestSignUpdate(hamlet_1) failed.\n");
+        goto cleanup;
+    }
+    if (EVP_DigestSignUpdate(sign_context, hamlet_2, strlen(hamlet_2)) != 1) {
+        fprintf(stderr, "EVP_DigestSignUpdate(hamlet_2) failed.\n");
+        goto cleanup;
+    }
+    /* Call EVP_DigestSignFinal to get signature length sig_len */
+    if (EVP_DigestSignFinal(sign_context, NULL, &sig_len) != 1) {
+        fprintf(stderr, "EVP_DigestSignFinal failed.\n");
+        goto cleanup;
+    }
+    if (sig_len <= 0) {
+        fprintf(stderr, "EVP_DigestSignFinal returned invalid signature length.\n");
+        goto cleanup;
+    }
+    sig_value = OPENSSL_malloc(sig_len);
+    if (sig_value == NULL) {
+        fprintf(stderr, "No memory.\n");
+        goto cleanup;
+    }
+    if (EVP_DigestSignFinal(sign_context, sig_value, &sig_len) != 1) {
+        fprintf(stderr, "EVP_DigestSignFinal failed.\n");
+        goto cleanup;
+    }
+    *sig_out_len = sig_len;
+    *sig_out_value = sig_value;
+    for (j = 0; j < sig_len; j++)
+        fprintf(stdout, "%02x", sig_value[j]);
+    fprintf(stdout, "\n");
+    fprintf(stdout, "Signature calculated properly.\n");
+    result = 1;
+
+cleanup:
+    /* OpenSSL free functions will ignore NULL arguments */
+    if (!result)
+        OPENSSL_free(sig_value);
+    EVP_PKEY_free(priv_key);
+    EVP_MD_CTX_free(sign_context);
+    return result;
+}
+
+static int demo_verify(OSSL_LIB_CTX *libctx, const char *sig_name,
+                       size_t sig_len, unsigned char *sig_value)
+{
+    int result = 0, i = 0, public = 1;
+    const char *propq = NULL;
+    EVP_MD_CTX *verify_context = NULL;
+    EVP_PKEY *pub_key = NULL;
+
+    /*
+     * Make a verify signature context to hold temporary state
+     * during signature verification
+     */    
+    verify_context = EVP_MD_CTX_new();
+    if (verify_context == NULL) {
+        fprintf(stderr, "EVP_MD_CTX_new failed.\n");
+        goto cleanup;
+    }
+    /*Get public key*/
```

Comment:
We'd normally has spaces before/after the *:
```c
/* Get public key */
```

Reply:
Done in latest upload, thanks for your comments.